File: automation/SharpAutomation/Playwright/Pages/HomePage.cs
using Microsoft.Playwright;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SharpAutomation.Playwright.Pages
{
    public class HomePage
    {
        private readonly IPage _page;

        public HomePage(IPage page) => _page = page;

        private ILocator SearchInput => _page.Locator("#txtGlobalSearch");

        public async Task NavigateAsync()
        {
            await _page.GotoAsync("https://dou.ua/");
        }

        public async Task<SearchPage> SearchAsync(string query)
        {
            await SearchInput.ClickAsync();
            await SearchInput.FillAsync(query);
            await _page.Keyboard.PressAsync("Enter");

            return new SearchPage(_page);
        }
    }
}

File: automation/JSAutomation/Playwright/pages/homePage.js
export class HomePage {
    /**
     * @param {import('@playwright/test').Page} page
     */
    constructor(page) {
      this.page = page;
      this.searchInput = '#txtGlobalSearch';
    }
  
    async goto() {
      await this.page.goto('https://dou.ua/');
    }
  
    async search(text) {
      await this.page.fill(this.searchInput, text);
      await this.page.keyboard.press('Enter');
    }
  }
  